MEN-5465: added release schedule & LTS release info (#1731)
mendersoftware/mender-docs: 302.Release-information/01.Release-schedule/docs.md

---
title: Release schedule
taxonomy:
    category: docs
---

Our requirements for release quality are always the same, with only the length of support differing between non-LTS releases and LTS releases.

> LTS releases are maintained through patch releases for one year, whereas non-LTS releases get patch releases only in case an urgent vulnerability is found in-between minor releases.

Reviewer: ```suggestion
LTS releases are maintained through patch releases for one year, whereas non-LTS releases get patch releases only if an urgent vulnerability is found between minor releases.
```

> The lifecycle of an LTS release begins on the release date and ends one year later at the scheduled support end date. Keeping these dates in mind helps when deciding which version to adopt and when to upgrade to a new version. To ensure the optimal operation of Mender within a release lifecycle, the release components should use any released patch updates once they become available.

Reviewer: ```suggestion
The lifecycle of an LTS release begins on the release date and ends one year later at the scheduled support end date. Keeping these dates in mind helps decide which version to adopt and when to upgrade to a new version. To ensure the optimal operation of Mender within a release lifecycle, the release components should use any released patch updates once they become available.
```

##### Current LTS releases

<!-- the version number includes the patch release here, to get picked up by autoversion, but will be a minor version in the result -->
<!--AUTOVERSION: "LTS releases: %"/lts -->

> At this time we support the following LTS releases: 1.0.0.

Reviewer: ```suggestion
At this time, we support the following LTS releases: 1.0.0.
```

Why is it 1.0.0?

Author: Any matching versioning string would work (afaik there's no way to tell `autoversion` what version style to look for) so the next time autoversion is run this will produce `3.0` (or `3.0, 3.3` once 3.3 is out)...
maybe I should set it to 0.0.0 to make the disconnect from the release reality clearer 👌

#### Release Cadence

> New minor version releases are released every three months, either as a normal or LTS release. From Mender 3.3 on, even-numbered releases count as normal releases, and odd-numbered releases a re considered LTS releases. Therefore, we will release LTS minor releases every six months.

Reviewer 2: "**approximately** every three months", maybe..?

> New minor version releases are released every three months, either as a normal or LTS release. From Mender 3.3 on, even-numbered releases count as normal releases, and odd-numbered releases a re considered LTS releases. Therefore, we will release LTS minor releases every six months.

Reviewer 2: > From Mender 3.3 on, even-numbered releases count as normal releases, and odd-numbered releases a re considered LTS releases

I don't think we should make people used to this schema, as there are several things which could break it: What we just experienced with the unfixable release is probably rare, but more common is that we do a major bump. How about instead:

> New minor version releases are released every three months, either as a normal or LTS release. LTS minor releases are released every six months, but version numbers may vary depending on the content of the release. Check back here to learn which versions are part of an LTS release series.

> New minor version releases are released every three months, either as a regular or as an LTS release. From Mender 3.3 on, even numbered releases count as regular releases and odd numbered releases are considered LTS releases. This results in LTS minor releases being released every six months.

Reviewer: ```suggestion
New minor version releases are released every three months, either as a normal or LTS release. From Mender 3.3 on, even-numbered releases count as normal releases, and odd-numbered releases are considered LTS releases. Therefore, we will release LTS minor releases every six months.
```

> Patch releases for LTS releases are published at the same time as a regular release is scheduled, unless an urgent vulnerability is discovered. 

Reviewer: Patch releases for LTS releases are published simultaneously as a normal release is scheduled unless an urgent vulnerability is discovered.
